subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

```diff
@@ -0,0 +1,752 @@
+
+A dependency verification configuration is _global_: a single file is used to configure verification of the whole build.
+
+An easy way to get started is therefore to generate the minimal configuration for an existing build.
+
+[[sec:bootstrapping-verification]]
+=== Bootstrapping dependency verification
+
+It's worth mentioning that while Gradle can generate a dependency verification file for you, you should always check whatever Gradle generated for you because your build may _already_ contain compromised dependencies without you knowing about it.
+Please refer to the appropriate <<#sec:checksum-verification,checksum verification>> or <<#sec:signature-verification,signature verification>> section for more information.
+
+If you plan on using <<#sec:signature-verification,signature verification>>, please also read the <<sec:bootstrapping-signature-verification,corresponding section>> of the docs.
+
+Bootstrapping can either be used to create a file from the beginning, or also to _update_ an existing file with new information.
+Therefore, it's recommended to always use the same parameters once you started bootstrapping.
+
+The dependency verification file can be generated with the following CLI instructions:
+
+
+----
+gradle --write-verification-file sha256 help
+----
+
+The `write-verification-file` flag requires the list of <<#sec:checksum-verification,checksums>> that you want to generate or `pgp` for <<#sec:signature-verification,signatures>>.
+
+Executing this command line will cause Gradle to:
+
+* resolve all <<declaring_dependencies.adoc#sec:resolvable-consumable-configs,resolvable configurations>>, which includes:
+** configurations from the root project
+** configurations from all subprojects
+** configurations from `buildSrc`
+** included builds configurations
+** configurations used by plugins
+* download all artifacts discovered during resolution
+* compute the requested checksums and possibly verify signatures depending on what you asked
+* At the end of the build, generate the configuration file which will contain the inferred _verification metadata_
+
+As a consequence, the `verification-metadata.xml` file will be used in subsequent builds to verify dependencies.
+
+[WARNING]
+====
+There are dependencies that Gradle _cannot_ discover this way.
+In particular, you will notice that the CLI above uses the `help` task.
+If you don't specify any task, Gradle will automatically run the default task and generate a configuration file at the end of the build too.
+
+The difference is that Gradle _may_ discover more dependencies and artifacts depending on the tasks you execute.
+As a matter of fact, Gradle cannot discover automatically _detached configurations_, which are basically dependency graphs resolved as an internal implementation detail of the execution of a task: they are not, in particular, declared as an input of the task because they effectively depend on the configuration of the task at execution time.
```

Comment:
```suggestion
As a matter of fact, Gradle cannot automatically discover _detached configurations_, which are basically dependency graphs resolved as an internal implementation detail of the execution of a task: they are not, in particular, declared as an input of the task because they effectively depend on the configuration of the task at execution time.
```